services/movie_session.py

```diff
@@ -0,0 +1,65 @@
+from django.core.exceptions import ObjectDoesNotExist
+from db.models import MovieSession, Movie, CinemaHall
+from datetime import datetime
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    try:
+        movie = Movie.objects.get(id=movie_id)
+        cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    except ObjectDoesNotExist:
+        raise ValueError("Provided movie or cinema hall ID does not exist.")
+
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=movie,
+        cinema_hall=cinema_hall
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> MovieSession:
+    if session_date:
+        try:
+            date_obj = datetime.strptime(session_date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValueError("Invalid date format. Expected 'year-month-day'")
+        return MovieSession.objects.filter(show_time__date=date_obj)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
```

Comment:
The return type hint for `get_movie_session_by_id` should be `Optional[MovieSession]` instead of `MovieSession`, as the function may return `None` if no movie session is found with the given ID.